### PR TITLE
ci: override coq url in platform script

### PR DESCRIPTION
**Pull request by gares**
This PR passes to the platform script a flag to override the coq archive url.
Depends on https://github.com/coq/platform/pull/103

CC @MSoegtropIMC 

Closes #14204

### CODE REVIEW

**Comment by SkySkimmer** [comment]
https://gitlab.com/coq/coq/-/jobs/1226924009#L1441
>[coq.dev] synchronised from git+https://github.com/coq/coq.git#master

looks wrong

**Comment by gares** [comment]
Thanks, I need to make it percolate trough the bat file...

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -257,7 +257,8 @@ before_script:
   stage: stage-1
   interruptible: true
   variables:
-    PLATFORM: "https://github.com/coq/platform/archive/dev-ci.zip"
+    PLATFORM_SOURCES: "https://github.com/gares/platform/archive/override-pkg-url.zip"
+    PLATFORM_EXTRA_ARGS: "-override-dev-pkg=coq=${CI_PROJECT_URL}#${CI_COMMIT_SHA}"
```
Comment (SkySkimmer) [inline .gitlab-ci.yml]: you need to use windows variable syntax, not $foo
(the artifacts name below is similarly bugged but nothing relies on it)

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -257,7 +257,8 @@ before_script:
   stage: stage-1
   interruptible: true
   variables:
-    PLATFORM: "https://github.com/coq/platform/archive/dev-ci.zip"
+    PLATFORM_SOURCES: "https://github.com/gares/platform/archive/override-pkg-url.zip"
+    PLATFORM_EXTRA_ARGS: '"-override-dev-pkg=coq=${CI_PROJECT_URL}#${CI_COMMIT_SHA}"'
```
Comment (SkySkimmer) [inline .gitlab-ci.yml:261]: you need to use windows variable syntax, not $foo
(the artifacts name below is similarly bugged but nothing relies on it)

**Comment by gares** [comment]
Yes
```
[coq.dev] downloaded from https://gitlab.com/coq/coq#60d61551445b1f24fe747824f5c145fdc6a92107
[coqide.dev] downloaded from https://gitlab.com/coq/coq#60d61551445b1f24fe747824f5c145fdc6a92107
```

File: dev/ci/platform-windows.bat [hunk 3]
```diff
@@ -48,7 +48,10 @@ call coq_platform_make_windows.bat ^
   -extent=i ^
   -parallel=p ^
   -jobs=2 ^
-  -switch=d || GOTO ErrorCopyLogFilesAndExit
+  -switch=d ^
+  -override-dev-pkg="coq=%CI_PROJECT_URL%/-/archive/%CI_COMMIT_SHA%/coq-%CI_COMMIT_SHA%.tar.gz" ^
+  -override-dev-pkg="coqide=%CI_PROJECT_URL%/-/archive/%CI_COMMIT_SHA%/coqide-%CI_COMMIT_SHA%.tar.gz" ^
```
Comment (SkySkimmer) [inline dev/ci/platform-windows.bat:53]: shouldn't that be
```suggestion
  -override-dev-pkg="coqide=%CI_PROJECT_URL%/-/archive/%CI_COMMIT_SHA%/coq-%CI_COMMIT_SHA%.tar.gz" ^
```
?

Comment (gares) [reply dev/ci/platform-windows.bat:53]: ok, but I'll wait for CI to run before pushing a new commit

**Comment by gares** [comment]
>     7	/bin/sh: _build/install/default/bin/coqdep.exe: Argument list too long

OK, we are now blocked by another issue

**Comment by SkySkimmer** [comment]
see https://github.com/coq/coq/pull/14240

**Comment by gares** [comment]
that would do too.

**Comment by gares** [comment]
maybe I should rebase on top of https://github.com/coq/coq/pull/14240 and see if this fully works

**Comment by SkySkimmer** [comment]
Looks like it works

**Comment by ejgallego** [comment]
Needs a rebase as #14240 has been merged

**Comment by ejgallego** [comment]
> Needs a rebase as #14240 has been merged

Rebase still neeeded, despite the bot saying otherwise. @gares can you push and we merge soon so CI is fixed? Thanks!

**Comment by gares** [comment]
In order to remove the [hack] commit we need it to be merged into the platform

**Comment by ejgallego** [comment]
> In order to remove the [hack] commit we need it to be merged into the platform

Ok thanks, I had kinda understood that it was OK to live with that commit for a while.

**Comment by gares** [comment]
I merged the PR in the coq platform, and removed the hack here. If it works it can be merged.

**Review by ejgallego — APPROVED** [review]
Thanks @gares

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**